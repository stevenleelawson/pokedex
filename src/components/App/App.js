import React, { Component } from 'react';
import './App.css';
import FakeContainer from '../../containers/FakeContainer/'
import { connect } from 'react-redux';
import * as actions from '../../actions'
import loadingGif from './loading-gif.gif';
import * as api from '../../apiCalls';
import PokeContainer from '../../containers/PokeContainer/PokeContainer';

export class App extends Component {
  async componentDidMount () {
    const pokeTypes =  await api.getTypes()
    this.props.loadPokemonTypes(pokeTypes)
    console.log('pokes',this.props.types)
  }

  render() {
    return (
      <div className='App'>
        <h1 className='header'> POKÃ©DEX </h1>
        {this.props.types.length === 0 &&
          <img src={loadingGif} />
        }
        <PokeContainer />
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  types: state.types
})

export const mapDispatchToProps = (dispatch) => ({
  loadPokemonTypes: (pokemonType) => dispatch(actions.loadPokemonTypes(pokemonType))
})

export default connect(mapStateToProps, mapDispatchToProps)(App);
