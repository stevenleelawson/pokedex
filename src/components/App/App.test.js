import React from 'react';
import {App} from './App';
import { shallow } from 'enzyme';

describe('App', () => {

  let wrapper;
  beforeEach(() => {
    let mockTypes = [{id: 1, name: 'normal'}]
    wrapper = shallow(<App types={mockTypes}/>);
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
