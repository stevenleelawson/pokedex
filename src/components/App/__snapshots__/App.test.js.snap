// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    types={
      Array [
        Object {
          "id": 1,
          "name": "normal",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="header"
        >
           POKéDEX 
        </h1>,
        false,
        <Connect(PokeContainer) />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " POKéDEX ",
          "className": "header",
        },
        "ref": null,
        "rendered": " POKéDEX ",
        "type": "h1",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="header"
          >
             POKéDEX 
          </h1>,
          false,
          <Connect(PokeContainer) />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " POKéDEX ",
            "className": "header",
          },
          "ref": null,
          "rendered": " POKéDEX ",
          "type": "h1",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
